Mathlib/Algebra/Ring/Opposite.lean

```diff
@@ -150,10 +150,10 @@ instance nonUnitalRing [NonUnitalRing α] : NonUnitalRing αᵃᵒᵖ :=
   { AddOpposite.addCommGroup α, AddOpposite.semigroupWithZero α, AddOpposite.distrib α with }
 
 instance nonAssocRing [NonAssocRing α] : NonAssocRing αᵃᵒᵖ :=
-  { AddOpposite.addCommGroup α, AddOpposite.mulZeroOneClass α, AddOpposite.distrib α with }
+  { AddOpposite.addCommGroupWithOne α, AddOpposite.mulZeroOneClass α, AddOpposite.distrib α with }
 
 instance ring [Ring α] : Ring αᵃᵒᵖ :=
-  { AddOpposite.addCommGroup α, AddOpposite.monoid α, AddOpposite.semiring α with }
+  { AddOpposite.addCommGroupWithOne α, AddOpposite.monoid α, AddOpposite.semiring α with }
```

Comment:
These changes didn't exist in the Lean 3 version either.

Reply:
Yeah, this is all stuff that broke because of the way Lean 4 finds the instances. It's already wrong in mathlib but somehow we didn't hit the problem.